# <a id='toc1_'></a>[Calcul de chimie quantique: cas de la molécule de furan](#toc0_)

 **D. SHARON CABRELLE** sharon.djaparou@facsciences-uy1.cm
 
    * Master's degree in Physics - PMB - Quantum Optics
       
February 2025


# 

# INTRODUCTION

Dans le domaine de la chimie moderne, la compréhension approfondie des molécules et de leurs propriétés est essentielle pour le développement de nouvelles technologies et applications. Les molécules, en tant qu'unités fondamentales de la matière, jouent un rôle crucial dans divers domaines tels que la pharmacologie, les matériaux avancés, et la chimie environnementale. Cette présentation se concentre sur une molécule spécifique, en examinant son intérêt scientifique, ses applications pratiques, ainsi que ses propriétés à l'aide d'outils informatiques comme RDKit. En outre, des calculs de chimie quantique, notamment par les méthodes HF, DFT et post-HF, seront réalisés pour explorer les caractéristiques électroniques et géométriques de la molécule, en fournissant des analyses et des interprétations.

#  Description physique de la molécule de furan

Le furan est un composé organique hétérocyclique, ayant la formule chimique *C₄H₄O*. Il se compose d’un cycle à cinq atomes, dont quatre carbones, quatre atomes d'hydrogène et un oxygène, avec des doubles liaisons.  C'est également un liquide incolore à température ambiante, avec un point d’ébullition d’environ 31 °C. Il a une odeur douce et agréable, souvent comparée à celle des fruits.

# Intérêts et Applications de la molécule

### Chimie Organique et Synthèse
Le furan est une plateforme chimique importante pour la synthèse de divers composés. Il sert de précurseur dans la fabrication de produits chimiques plus complexes, notamment des polymères et des produits pharmaceutiques. Les réactions de cyclisation, telles que la réaction de Diels-Alder, permettent de créer des structures plus complexes à partir du furan.

### Industrie Pharmaceutique
De nombreux médicaments contiennent des unités furan dans leur structure. Le furan et ses dérivés ont montré des propriétés biologiques intéressantes, notamment des activités anti-inflammatoires, antitumorales et antimicrobiennes.

### Matériaux et Polymères
Le furan est utilisé dans la production de résines et de polymères, comme les résines phénoliques, qui sont largement utilisées dans la fabrication de matériaux composites et d'adhésifs. Les polymères à base de furan sont également étudiés pour leur durabilité et leur recyclabilité, offrant une alternative plus écologique aux plastiques traditionnels.

### Agriculture et Alimentation
Dans le domaine alimentaire, le furan peut se former lors de la cuisson de certains aliments. Bien que sa présence en tant que contaminant soit préoccupante, des études sont en cours pour comprendre ses effets et minimiser sa formation. Des recherches explorent également son utilisation comme intermédiaire dans la synthèse de pesticides.

### Bioénergie
Le furan et ses dérivés sont étudiés comme candidats potentiels pour la production de bioénergie et de biocarburants. Ils peuvent être produits à partir de biomasse, offrant une voie renouvelable pour des sources d'énergie alternatives.

#  Présentation de la molécule avec rdkit

In [35]:
# importation des bibiothèques neccessaires
import pyscf

pyscf.__version__

'2.7.0'

In [41]:
from pyscf import gto

In [42]:
with open('molecules_xyz.txt', 'r') as source_file:
    # Lire le contenu du fichier source ligne par ligne
    lines = source_file.readlines()
    
    # Extraire les lignes souhaitées
    extracted_lines = lines[69:79]  # Exemple : extraire les lignes 3 à 5
    
# Ouvrir le fichier de destination en écriture
with open('furan.txt', 'w') as destination_file:
    # Écrire les lignes extraites dans le fichier de destination
    for line in extracted_lines:
        destination_file.write(line)

####  Réprésentation de la molécule avec py3Dmol

In [43]:
with open('furan.txt', 'r') as file:
    lines = file.readlines()[1:] 
    mol_xyz = ''.join(lines).strip()

mol = gto.M(
    atom=mol_xyz, 
    basis="6-31g", 
    verbose=4,
    charge=0,      # 0 par défaut
    spin=0,        # 0 par défaut, défini comme (n_up - n_down)
    symmetry=True, # False par défaut
)

import py3Dmol
# 3D representation
xyz_view = py3Dmol.view(width=300,height=300)
xyz_view.addModel(mol.tostring(format="xyz"),'xyz')
xyz_view.setStyle({'stick':{}})
xyz_view.zoomTo()
xyz_view.show()

System: uname_result(system='Linux', node='sharon-Latitude-5480', release='6.8.0-49-generic', version='#49~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Nov  6 17:42:15 UTC 2', machine='x86_64')  Threads 4
Python 3.11.7 (main, Dec 15 2023, 18:12:31) [GCC 11.2.0]
numpy 1.26.4  scipy 1.11.4  h5py 3.9.0
Date: Sat Feb 15 15:51:55 2025
PySCF version 2.7.0
PySCF path  /home/sharon/anaconda3/lib/python3.11/site-packages/pyscf

[CONFIG] conf_file None
[INPUT] verbose = 4
[INPUT] num. atoms = 9
[INPUT] num. electrons = 36
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry True subgroup None
[INPUT] Mole.unit = angstrom
[INPUT] Symbol           X                Y                Z      unit          X                Y                Z       unit  Magmom
[INPUT]  1 O      1.101400000000   0.026700000000   0.304100000000 AA    2.081344353596   0.050455687526   0.574665714480 Bohr   0.0
[INPUT]  2 C      0.600900000000   0.115700000000  -0.976700000000 AA    1.135536428251   0.2

3Dmol.js failed to load for some reason. Please check your browser console for error messages.

In [39]:
# creation de l'objet mol. construit l'object molécule
mol.build()

####  Imprimons les différentes propriétés physico-chimique de la molécule

In [40]:
print(f'Le nombre total d\'électrons est {mol.nelectron} et le nombre total d\'électrons (alpha, béta) est {mol.nelec}')
print(f'Le nombre d\'orbitales atomiques, dans la base {mol.basis}, est {mol.nao_nr()}')
print(f'L\'énergie nucléaire vaut {mol.energy_nuc()} Hartrees')

Le nombre total d'électrons est 36 et le nombre total d'électrons (alpha, béta) est (18, 18)
Le nombre d'orbitales atomiques, dans la base 6-31g, est 53
L'énergie nucléaire vaut 159.42412426803912 Hartrees


In [48]:
from rdkit import Chem
smiles = " [2H]C1(C(C(OC1([2H])[2H])([2H])[2H])([2H])[2H])[2H]" 
mol_rdkit = Chem.MolFromSmiles(smiles)

In [74]:
import os
import pandas as pd
from rdkit.Chem import Descriptors
properties = ['Poids_Mol','poids_mol_moy', 'LogP', 'TPSA','hbd','hba']

# Calcul des descripteurs
#poid_moleculaire
molwt = Descriptors.MolWt(mol_rdkit)
#poid moleculaire moyen
molwtE = Descriptors.ExactMolWt(mol_rdkit)
# solubilite
logp = Descriptors.MolLogP(mol_rdkit)
#polarite
tpsa = Descriptors.TPSA(mol_rdkit)
#nombre de liasons donneurs d'H
hbd = Descriptors.NumHDonors(mol_rdkit)
#nombre de liason accepteurs d'H
hba = Descriptors.NumHAcceptors(mol_rdkit)

# Ajout des propriétés à la liste de données
data=[molwt , molwtE, logp, tpsa,hbd,hba]
data1=[68.07 , 68.075  ,1.34, 13.14 ,0,1]

df_HF = pd.DataFrame({"properties ":properties, 
                      'valeurs':data,
                      "valeurs_pupchem":data1})
df_HF.set_index("properties ", inplace=True)### <a id='toc1_3_'></a>[CONSIGNE](#toc0_)

df_HF

valeurs  valeurs_pupchem
properties                               
Poids_Mol      80.155814           68.070
poids_mol_moy  80.107729           68.075
LogP            0.796800            1.340
TPSA            9.230000           13.140
hbd             0.000000            0.000
hba             1.000000            1.000

Les informations présentes dans le DataFrame pandas sont: 

* `Poids_Mol` : Il s'agit du `poids moléculaire` de la molécule, exprimé en grammes par mole `(g/mol)`. C'est la masse totale de la molécule qui est de `80.155814	g/mol` qui est equivalent a celle de `pubchem` estimé a  `68.070 g/mol`

* `poids_mol_moy` : Il s'agit du `poids moléculaire moyen`, calculé en prenant en compte la distribution isotopique des atomes donc la valeurs est de `80.107729g\mol` . qui est légèrement différent du poids moléculaire brute.

* `LogP` : Il s'agit du coefficient de partage `octanol/eau (logP)` de la molécule. Un coefficient de partage `octanol/eau (logP)` de `0.796800	` suggère une légère `hydrophobicité`(tendance d'une substance à repousser ou à éviter l'eau) pour la molécule. Qui est l'aptitude de la molécule à traverser les membranes biologiques

* `TPSA` : Il s'agit de la surface polaire totale `Total Polar Surface Area)` de la molécule, exprimée en `Ų (angstroms carrés)`. C'est une mesure de la surface de la molécule qui est polaire et capable d'interactions avec des solvants polaires sa valeurs ici est de `12.89Ų`.

* `hbd` : Il s'agit du nombre de donneurs de liaisons hydrogène `(hydrogen bond donors)` dans la molécule. Cela compte le nombre d'atomes d'hydrogène liés à des atomes d'oxygène, d'azote ou de soufre, qui peuvent former des liaisons hydrogène étant donné que la dans le cas de la molécule `1,3-Thiazole` les atomes d'hydrogènes  n'ont que des liaisons  avec les atomes de carbones en conséuent on a une valeurs `hba= 0`.

* `hba` : Il s'agit du nombre d'accepteurs de liaisons hydrogène `(hydrogen bond acceptors)` dans la molécule. Cela compte le nombre d'atomes d'oxygène, d'azote ou de soufre dans la molécule, qui peuvent accepter des liaisons hydrogène,dans notre molécule nous uniquement un atome de soufre et un atome d'azote d'ou la valeur de `hba=2`.

##  Calcul de l'énergie en utilisant la DFT

In [75]:
# Initialize the dictionary to store the energies and reference values
from pyscf import dft
import time

start = time.time()
mdf=dft.RKS(mol, xc="B3LYP").run()
mdf_time1 = time.time() - start



******** <class 'pyscf.dft.rks.RKS'> ********
method = RKS
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmpi5slfp7b
max_memory 4000 MB (current use 485 MB)
XC library pyscf.dft.libxc version 6.1.0
    S. Lehtola, C. Steigemann, M. J.T. Oliveira, and M. A.L. Marques.,  SoftwareX 7, 1–5 (2018)
XC functionals = B3LYP
    P. J. Stephens, F. J. Devlin, C. F. Chabalowski, and M. J. Frisch.,  J. Phys. Chem. 98, 11623 (1994)
radial grids: 
    Treutler-Ahlrichs [JCP 102, 346 (1995); DOI:10.1063/1.469408] (M4) radial grids
    
becke partition: Becke, JCP 88, 2547 (1988); DOI:10.1063/1.454033
pruning grids: <function nwchem_prune at 0x7a0b3c127a60>
grids dens level: 3
symmetrized grids: True
atomic radii adjust function: <function treutler_atomic_ra

In [76]:
# Initialize the dictionary to store the energies and reference values
from pyscf import dft

start = time.time()
mdf=dft.RKS(mol, xc="PBE").run()
mdf_time2 = time.time() - start



******** <class 'pyscf.dft.rks.RKS'> ********
method = RKS
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmp_0ws6aca
max_memory 4000 MB (current use 486 MB)
XC library pyscf.dft.libxc version 6.1.0
    S. Lehtola, C. Steigemann, M. J.T. Oliveira, and M. A.L. Marques.,  SoftwareX 7, 1–5 (2018)
XC functionals = PBE
    J. P. Perdew, K. Burke, and M. Ernzerhof.,  Phys. Rev. Lett. 77, 3865 (1996)
    J. P. Perdew, K. Burke, and M. Ernzerhof.,  Phys. Rev. Lett. 78, 1396 (1997)
    J. P. Perdew, K. Burke, and M. Ernzerhof.,  Phys. Rev. Lett. 77, 3865 (1996)
    J. P. Perdew, K. Burke, and M. Ernzerhof.,  Phys. Rev. Lett. 78, 1396 (1997)
radial grids: 
    Treutler-Ahlrichs [JCP 102, 346 (1995); DOI:10.1063/1.469408] (M4) radial grids
    
becke pa

In [77]:
# Initialize the dictionary to store the energies and reference values
from pyscf import dft

start = time.time()
mdf=dft.RKS(mol, xc="TPSS").run()
mdf_time3 = time.time() - start



******** <class 'pyscf.dft.rks.RKS'> ********
method = RKS
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmpk5q0iyqx
max_memory 4000 MB (current use 485 MB)
XC library pyscf.dft.libxc version 6.1.0
    S. Lehtola, C. Steigemann, M. J.T. Oliveira, and M. A.L. Marques.,  SoftwareX 7, 1–5 (2018)
XC functionals = TPSS
    J. Tao, J. P. Perdew, V. N. Staroverov, and G. E. Scuseria.,  Phys. Rev. Lett. 91, 146401 (2003)
    J. P. Perdew, J. Tao, V. N. Staroverov, and G. E. Scuseria.,  J. Chem. Phys. 120, 6898 (2004)
    J. Tao, J. P. Perdew, V. N. Staroverov, and G. E. Scuseria.,  Phys. Rev. Lett. 91, 146401 (2003)
    J. P. Perdew, J. Tao, V. N. Staroverov, and G. E. Scuseria.,  J. Chem. Phys. 120, 6898 (2004)
radial grids: 
    Treutler-Ahlrichs 

In [83]:
import os
import pandas as pd
# Définir les énergies et les temps de calcul pour chaque fonctionnelle DFT
energies = {
    'Fonctionnelle': ['B3LYP', 'PBE', 'TPSS'],
    'Énergie (Hartree)': [-229.951150579556, -229.680409971373, -229.99751491233],
    'Temps de calcul (s)': [mdf_time1, mdf_time2, mdf_time3]  # Remplacez mdf_time par les valeurs réelles
}

dft_energies = pd.DataFrame({"properties ":properties, 
                      'valeurs':data,
                      "valeurs_pupchem":data1})
dft_energies.set_index("properties ", inplace=True)
# Créer un DataFrame à partir du dictionnaire
dft_energies = pd.DataFrame(energies)

# Afficher le tableau
print(dft_energies)

  Fonctionnelle  Énergie (Hartree)  Temps de calcul (s)
0         B3LYP        -229.951151             5.592599
1           PBE        -229.680410             6.084183
2          TPSS        -229.997515             8.791888


## RHF

In [79]:
from pyscf import scf 
from pyscf.data.nist import HARTREE2EV as au2ev 
import pandas as pd

In [71]:
mf = scf.RHF(mol) # Create a PySCF HF mean-field object
mf.x2c().density_fit() # To speed up calculations
mf.kernel()



******** <class 'pyscf.scf.hf.RHF'> ********
method = RHF
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmpy0f4fzdz
max_memory 4000 MB (current use 485 MB)
Set gradient conv threshold to 3.16228e-05
Initial guess from minao.
init E= -229.893458838602
  HOMO = -0.178597272652802  LUMO = 0.0711546430549261
cycle= 1 E= -228.461538337727  delta_E= 1.43  |g|= 0.419  |ddm|= 2.89
  HOMO = -0.357654665435535  LUMO = 0.129753328497563
cycle= 2 E= -228.512479066367  delta_E= -0.0509  |g|= 0.197  |ddm|= 0.714
  HOMO = -0.323291264429016  LUMO = 0.150466700495064
cycle= 3 E= -228.520621682751  delta_E= -0.00814  |g|= 0.0629  |ddm|= 0.213
  HOMO = -0.319758497208221  LUMO = 0.15709580726888
cycle= 4 E= -228.521644969222  delta_E= -0.00102  |g|= 0.0146  |

-228.52172268564095

## Post HF

In [13]:
import time
from pyscf import scf


In [14]:
start = time.time()
myhf = scf.RHF(mol).run()
myhf_time = time.time() - start 



******** <class 'pyscf.scf.hf.RHF'> ********
method = RHF
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmpu0wtgug2
max_memory 4000 MB (current use 257 MB)
Set gradient conv threshold to 3.16228e-05
Initial guess from minao.
init E= -229.893458838602
  HOMO = -0.178597272652799  LUMO = 0.0711546430549282
cycle= 1 E= -228.461538337727  delta_E= 1.43  |g|= 0.419  |ddm|= 2.89
  HOMO = -0.357654665435532  LUMO = 0.129753328497564
cycle= 2 E= -228.512479066367  delta_E= -0.0509  |g|= 0.197  |ddm|= 0.714
  HOMO = -0.323291264429021  LUMO = 0.150466700495057
cycle= 3 E= -228.52062168275  delta_E= -0.00814  |g|= 0.0629  |ddm|= 0.213
  HOMO = -0.319758497208216  LUMO = 0.157095807268879
cycle= 4 E= -228.521644969222  delta_E= -0.00102  |g|= 0.0146  |

In [15]:
from pyscf import mp
start = time.time()
mymp2 = mp.MP2(myhf).run()
mymp2_time = time.time() - start +  myhf_time
energie1=mymp2.e_tot
print(energie1)


******** <class 'pyscf.mp.mp2.MP2'> ********
nocc = 18, nmo = 53
max_memory 4000 MB (current use 275 MB)
E(MP2) = -229.004293590945  E_corr = -0.482570905304997
E(SCS-MP2) = -228.995452007022  E_corr = -0.473729321382485
E_corr(same-spin) = -0.121564344211744
E_corr(oppo-spin) = -0.361006561093253
-229.00429359094494


#### <a id='toc1_1_'></a>2.2.3[Calcule du  CCSD avec CC](#toc0_)
* La [CC(Coupled_cluster)](https://en.wikipedia.org/wiki/Coupled_cluster) est une méthode perturbative qui améliore l'approximation de Hartree-Fock.

In [16]:
from pyscf import cc

start = time.time()
mycc = cc.CCSD(myhf).run()
mycc_time = time.time() - start +myhf_time
energie2=mycc.e_tot
print(energie2)


******** <class 'pyscf.cc.ccsd.CCSD'> ********
CC2 = 0
CCSD nocc = 18, nmo = 53
max_cycle = 50
direct = 0
conv_tol = 1e-07
conv_tol_normt = 1e-05
diis_space = 6
diis_start_cycle = 0
diis_start_energy_diff = 1e+09
max_memory 4000 MB (current use 286 MB)


<class 'pyscf.cc.ccsd.CCSD'> does not have attributes  converged


Init t2, MP2 energy = -229.004293604105  E_corr(MP2) -0.482570918464576
Init E_corr(CCSD) = -0.482570918467117
cycle = 1  E_corr(CCSD) = -0.487035903329988  dE = -0.00446498486  norm(t1,t2) = 0.0722545
cycle = 2  E_corr(CCSD) = -0.504352445029364  dE = -0.0173165417  norm(t1,t2) = 0.0268194
cycle = 3  E_corr(CCSD) = -0.506272390726049  dE = -0.0019199457  norm(t1,t2) = 0.0132836
cycle = 4  E_corr(CCSD) = -0.510032159521885  dE = -0.0037597688  norm(t1,t2) = 0.00633887
cycle = 5  E_corr(CCSD) = -0.510124156500411  dE = -9.19969785e-05  norm(t1,t2) = 0.00135032
cycle = 6  E_corr(CCSD) = -0.509975228448143  dE = 0.000148928052  norm(t1,t2) = 0.000515884
cycle = 7  E_corr(CCSD) = -0.509966803586925  dE = 8.42486122e-06  norm(t1,t2) = 0.000180114
cycle = 8  E_corr(CCSD) = -0.509978326673995  dE = -1.15230871e-05  norm(t1,t2) = 7.27922e-05
cycle = 9  E_corr(CCSD) = -0.509975217418459  dE = 3.10925554e-06  norm(t1,t2) = 3.04833e-05
cycle = 10  E_corr(CCSD) = -0.50997582391564  dE = -6.0649718

#### <a id='toc1_1_'></a>[ Calcule du  CCSD(T) avec CCSD](#toc0_)

In [17]:
start = time.time()
e_ccsd_t = mycc.ccsd_t()
mycct_time = time.time() - start + myhf_time

CCSD(T) correction = -0.0179409187665789


In [18]:
mf_times = [mdf_time, mymp2_time, mycc_time, mycct_time]
energies = [mdf.e_tot,energie1,energie2,energie2+e_ccsd_t]
ref_energies=[-569.077723,-568.010849,-568.030302	,-568.069584]
ecart_energies=[(mdf.e_tot + 569.077723)*627.5095,(energie1 + 569.077723)*627.5095,(energie2 + 569.077723)*627.5095,(energie2+e_ccsd_t + 569.077723)*627.5095]
mf_times

[4.494778156280518, 1.5649034976959229, 13.002567052841187, 3.488703727722168]

In [84]:
import pandas as pd

# Définir les énergies et les temps de calcul pour chaque méthode
energies = {
    'Méthode': ['B3LYP (DFT)', 'PBE (DFT)', 'TPSS (DFT)', 'MP2', 'CCSD', 'CCSD(T)'],
    'Énergie (Hartree)': [
        -229.951150579556,  # B3LYP
        -229.680409971373,  # PBE
        -229.99751491233,   # TPSS
        -229.004293590945,  # MP2
        -229.0316979788074, # CCSD
        -229.0316979788074 + (-0.0179409187665789)  # CCSD(T)
    ],
    'Temps de calcul (s)': [
        107.79,             # B3LYP (remplacez par le temps réel)
        mdf_time,           # PBE (remplacez par le temps réel)
        mdf_time,           # TPSS (remplacez par le temps réel)
        mymp2_time,         # MP2 (remplacez par le temps réel)
        mycc_time,         # CCSD (remplacez par le temps réel)
        mycct_time         # CCSD(T) (remplacez par le temps réel)
    ]
}

# Créer un DataFrame à partir du dictionnaire
df_energies = pd.DataFrame(energies)

# Afficher le tableau
print(df_energies)

       Méthode  Énergie (Hartree)  Temps de calcul (s)
0  B3LYP (DFT)        -229.951151           107.790000
1    PBE (DFT)        -229.680410             7.930412
2   TPSS (DFT)        -229.997515             7.930412
3          MP2        -229.004294             1.564903
4         CCSD        -229.031698            13.002567
5      CCSD(T)        -229.049639             3.488704


In [ ]:
# Create the results dataframe
import pandas as pd
# Define the calculation methods
methods = ['DFT', 'MP2', 'CCSD', 'CCSD(T)']
df_HF = pd.DataFrame({"Methodes":methods, 
                      'Total energy':energies,
                      "Reference energy":ref_energies,
                      "ecart_ref_tot(kcal/mol)":ecart_energies,
                      "Time":mf_times})

df_HF

# Conclusion

L'étude approfondie de la molécule choisie révèle non seulement ses propriétés fondamentales mais aussi son potentiel d'application dans divers secteurs. Les analyses effectuées à l'aide de RDKit et les calculs de chimie quantique, bien que nécessitant des ressources informatiques significatives, offrent des informations précieuses qui complètent les données expérimentales disponibles sur des plateformes telles que PubChem et PDB. En résumé, cette démarche permet d'établir un lien entre la théorie et la pratique, tout en fournissant une base solide pour de futures recherches et développements dans le domaine de la chimie moléculaire.